src/main.rs

```diff
@@ -307,7 +307,9 @@ fn matches_desired_prefix(hash: &[u8; SHA1_BYTE_LENGTH], prefix: &HashPrefix) ->
 
 fn create_git_object_file(search_result: &HashMatch) -> io::Result<()> {
     let compressed_object = zlib_compress(&search_result.data)?;
-    let dir_path = format!(".git/objects/{:02x}", search_result.hash[0]);
+    let git_dir_bytes = run_command("git", &["rev-parse", "--git-dir"]);
+    let git_dir = &String::from_utf8(git_dir_bytes).expect("git rev-parse --git-dir returned invalid utf8");
```

Comment:
After trying this out locally it seems like the example code that I gave has a bug; `git rev-parse --git-dir` returns output with a trailing newline which ends up included in the filename. So the new created file is called `.git\n/objects/...`, which is interpreted as a regular file by git since it's not in the `.git` directory. This also might explain the error you were getting in https://github.com/not-an-aardvark/lucky-commit/issues/4#issuecomment-396152850.

To fix this, I think one option would be to use something like

```rust
let git_dir = String::from_utf8(git_dir_bytes).expect("git rev-parse --git-dir returned invalid utf8")[..-1];
```

I haven't tested this locally though, so I would recommend trying it out beforehand to double-check that it works. (You can do something like `./target/release/lucky_commit 000` to only customize the first three characters of the hash, which should still be good enough to verify that it's working without needing to wait a long time.)

Reply:
"the trait `std::ops::Neg` is not implemented for `usize`"